test/integration/compose/images.spec.ts

```diff
@@ -194,6 +124,10 @@ describe('compose/images', () => {
 			),
 		];
 
+		// INFO: we cannot create specific references to test on the engine, we need to
+		// use mockerode instead.
+		// QUESTION: Maybe the image search is overspecified and we should find a
+		// common identifier for all image search (e.g. label?)
```

Comment:
This is more of an architecture question, we are querying images by id, digest, tag, full name. And that also affects testing because there is some of that stuff that we cannot emulate on the engine.

I've had the idea for a long time that the supervisor could build (instead of push) a new image based on the image from the target state and add some metadata there. A label with the digest be part of that metadata and that would be a unique identifier that the supervisor could look for. This would also allow us to get rid of the image database as all the data we need would be in the engine.

Reply:
Sounds interesting, I remember you mentioned this before. Let's leave the `QUESTION` comment in then as a hint for this possible future feature.